### ISSUE TITLE
Compile error after reformatting inline closure in guard 

### ISSUE THREAD

**Issue by benkay** [nicklockwood/SwiftFormat#1556]
We had this (questionable) code

```swift
guard let foo = {
    if condition {
        return bar()
    } else {
        return baz()
    }
}() else {
    return nil
}
```

0.52.8 reformats this to
```swift
guard let foo = if condition {
    return bar()
} else {
    return baz()
} else {
    return nil
}
```

Which fails to compile (and if it did, would have different semantics).

I suppose this should be formatted as

```swift
guard let foo = {
    if condition {
       bar()
    } else {
       baz()
    }
}() else {
    return nil
}
```

Although we actually hand-changed this to the more sensible

```swift
let foo = if condition { bar() } else { baz() }
guard let foo else {
    return nil
}
```

**Comment by nicklockwood** [comment]
@benkay thanks for reporting this. It seems like you'd found a a good solution, but I'll get it fixed asap anyway.

**Comment by nicklockwood** [comment]
@calda am I right in thinking the bug here is in the `redundantClosure` rule? Specifically that closures containing `if` or `guard` expressions should not be removed when inside a conditional expression?

I think the simplest test case is probably something like:

```swift
guard let foo = {
  if condition {
    bar()
  }
}() else {
  return
}
```

**Comment by calda** [comment]
Yeah. I'm surprised the changes in https://github.com/nicklockwood/SwiftFormat/pull/1553 don't prevent this though. Will take a look.

**Comment by nicklockwood** [comment]
@calda I've just pushed a fix to develop.

**Comment by calda** [comment]
You code change looks reasonable to me

**Comment by nicklockwood** [comment]
@benkay fixed in 0.52.9
